Sources/PackageLoading/PackageBuilder.swift

```diff
@@ -836,19 +836,26 @@ public final class PackageBuilder {
     }
 
     /// Returns the list of platforms supported by the manifest.
-    func platforms() -> [SupportedPlatform] {
-        if let platforms = _platforms {
+    func platforms(isTest: Bool = false) -> [SupportedPlatform] {
+        if let platforms = _platforms[isTest] {
             return platforms
         }
 
         var supportedPlatforms: [SupportedPlatform] = []
 
         /// Add each declared platform to the supported platforms list.
         for platform in manifest.platforms {
+            let declaredPlatform = platformRegistry.platformByName[platform.platformName]!
+            var version = PlatformVersion(platform.version)
+
+            // macOS tests have a higher minimum deployment target
```

Comment:
Can you a more detailed note on why we have to special case here?

Reply:
> > How about we leave it at a fixed 10.15 for now?
> 
> I think we can actually do the same thing you suggested for XCTest and use the deployment target of libPackageDescription.

Hmm, yeah, that seems like a good idea!